src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3392,5 +3363,90 @@ private void EmitBox(TypeSymbol type, SyntaxNode syntaxNode)
             _builder.EmitOpCode(ILOpCode.Box);
             EmitSymbolToken(type, syntaxNode);
         }
+
+        private void EmitCalli(BoundFunctionPointerInvocation ptrInvocation, UseKind useKind)
+        {
+            EmitExpression(ptrInvocation.InvokedExpression, used: true);
```

Comment:
lol somehow I completely missed that we were evaluating the receiver expression for the first time here.